types/sip.js/index.d.ts

```diff
@@ -3,331 +3,325 @@
-        authorizationUser?: string;
-        autostart?: boolean;
-        connectionRecoveryMaxInterval?: number;
-        connectionRecoveryMinInterval?: number;
-        displayName?: string;
-        hackCleanJitsiSdpImageattr?: boolean;
-        hackStripTcp?: boolean;
-        hackIpInContact?: boolean;
-        hackViaTcp?: boolean;
-        hackWssInTransport?: boolean;
-        iceCheckingTimeout?: number;
-        instanceId?: string;
-        log?: {
-            builtinEnabled?: boolean;
-            level?: number | string;
-            connector?(level: string, category: string, label: string, content: string): void;
-        };
-        mediaHandlerFactory?: WebRTC.MediaHandlerFactory;
-        noAnswerTimeout?: number;
-        password?: string;
-        register?: boolean;
-        registerExpires?: number;
-        registrarServer?: string;
-        rel100?: string;
-        replaces?: string;
+export interface RenderHint {
+    remote?: Element;
+    local?: Element;
+}
+
+export interface MediaConstraints {
+    audio: boolean;
+    video: boolean;
+}
+
+export interface TurnServer {
+    urls?: string | string[];
+    username?: string;
+    password?: string;
+}
+
+export namespace WebRTC {
+    interface Options {
         stunServers?: string | string[];
-        traceSip?: boolean;
         turnServers?: TurnServer | TurnServer[];
-        usePreloadedRoute?: boolean;
-        userAgentString?: string;
-        wsServerMaxReconnection?: number;
-        wsServerReconnectionTimeout?: number;
+        RTCConstraints?: any;
     }
 
-    /* Options */
-    interface ExtraHeadersOptions {
-        extraHeaders?: string[];
-    }
+    type MediaHandlerFactory = (session: Session, options: Options) => MediaHandler;
 
-    interface UnregisterOptions extends ExtraHeadersOptions {
-        all?: boolean;
+    class MediaHandler {
+        getLocalStreams(): any[];
+        getRemoteStreams(): any[];
+        render(renderHint: RenderHint): void;
+
+        on(name: 'userMediaRequest', callback: (constraints: MediaConstraints) => void): void;
+        on(name: 'addStream' | 'userMedia', callback: (stream: any) => void): void;
+        on(name: 'userMediaFailed', callback: (error: string) => void): void;
+        on(name: 'iceCandidate', callback: (candidate: any) => void): void;
+        on(
+            name: 'iceGathering' | 'iceGatheringComplete' | 'iceConnection' | 'iceConnectionChecking' | 'iceConnectionConnected' | 'iceConnectionCompleted' | 'iceConnectionFailed' |
+                  'iceConnectionDisconnected' | 'iceConnectionClosed' | string,
+            callback: () => void): void;
+        on(name: 'dataChannel' | 'getDescription' | 'setDescription', callback: (sdpWrapper: { type: string, sdp: string }) => void): void;
     }
+}
 
-    interface MessageOptions extends ExtraHeadersOptions {
-        contentType?: string;
-    }
+/* Parameters */
+export interface ConfigurationParameters {
```

Comment:
There are plenty of areas where bindings are incomplete, but I think this is not relevant to the current problem.